src/DiffSharp.Core/Model.fs

```diff
@@ -80,15 +80,28 @@ type ParameterDict() =
     /// <summary>TBD</summary>
     member d.iter(f:string*Parameter->unit) = for KeyValue(n, p) in d.values do f(n,p)
 
-    /// <summary>TBD</summary>
+    /// <summary>
+    ///  Adjust the parameters to include support for forward-mode automatic differentiation.
+    /// </summary>
+    /// <param name="derivatives">The derivatives of the parameters</param>
+    /// <param name="tag">The level tag for nested differentiation.  Defaults to the current global nesting level</param>
+    /// <remarks>
+    ///  After this call the current parameters of the model will have attached derivatives for forward mode propagation.
+    /// </remarks>
     member d.forwardDiff(derivatives:ParameterDict, ?tag:uint32) = 
-        let tag = defaultArg tag GlobalNestingLevel.Current
```

Comment:
Note the default is applied in `p.forwardDiff`

Reply:
I think the reason for this line was to ensure all tensors within the ParameterDict will get the same tag, which is absolutely crucial. I think your change still preserves that in a single-threaded setting but it feels less safe.

Would you mind if I revert this with a comment to explain the reason?

Reply:
Yes that's fine